core/domain/classifier_services_test.py

```diff
@@ -234,28 +230,42 @@ def test_update_of_classifier_training_jobs(self):
             }
         ]
         # Job does not exist yet.
-        job_id = classifier_services.save_classifier_training_job(
+        job_id = classifier_services.create_classifier_training_job(
             feconf.INTERACTION_CLASSIFIER_MAPPING['TextInput']['algorithm_id'],
-            exp_id, 1, state_name, status, training_data)
+            interaction_id, exp_id, 1, state_name, training_data,
+            feconf.TRAINING_JOB_STATUS_NEW)
         classifier_training_job = (
             classifier_services.get_classifier_training_job_by_id(job_id))
         self.assertEqual(classifier_training_job.exp_id, exp_id)
-        self.assertEqual(classifier_training_job.status, status)
+        self.assertEqual(classifier_training_job.status,
+                         feconf.TRAINING_JOB_STATUS_NEW)
         classifier_training_job.update_status(test_status)
         # Updating existing job.
-        classifier_services.save_classifier_training_job(
-            classifier_training_job.algorithm_id,
-            classifier_training_job.exp_id,
-            classifier_training_job.exp_version,
-            classifier_training_job.state_name,
-            classifier_training_job.status,
-            classifier_training_job.training_data,
-            classifier_training_job.job_id)
+        classifier_services.update_classifier_training_job(
+            classifier_training_job)
         classifier_training_job = (
             classifier_services.get_classifier_training_job_by_id(job_id))
         self.assertEqual(classifier_training_job.exp_id, exp_id)
         self.assertEqual(classifier_training_job.status, test_status)
 
+    def test_mark_training_job_complete(self):
+        """Test the mark_training_job_complete method."""
+        exp_id = u'1'
+        state_name = 'Home'
+        interaction_id = 'TextInput'
+
+        job_id = classifier_services.create_classifier_training_job(
+            feconf.INTERACTION_CLASSIFIER_MAPPING[interaction_id][
+                'algorithm_id'], interaction_id, exp_id, 1, state_name,
+            [], feconf.TRAINING_JOB_STATUS_NEW)
+
+        classifier_services.mark_training_job_complete(job_id)
+
+        classifier_training_job = (
```

Comment:
To be more robust, repeat this block of code before marking the training job complete and verify that the status is _not_ marked as complete. Then you know that mark_training_job_complete() is the one that's actually responsible for the change.